Mathlib/Order/WithBot.lean

```diff
@@ -177,13 +179,23 @@ section LE
 
 variable [LE α] {x y : WithBot α}
 
-instance (priority := 10) le : LE (WithBot α) :=
-  ⟨fun o₁ o₂ => ∀ a : α, o₁ = ↑a → ∃ b : α, o₂ = ↑b ∧ a ≤ b⟩
+/-- The order on `WithBot α`, defined by `⊥ ≤ ⊥`, `⊥ ≤ ↑a` and `a ≤ b → ↑a ≤ ↑b`. -/
+@[mk_iff] -- TODO: `mk_iff` generates `lE_iff`
+protected inductive LE : WithBot α → WithBot α → Prop
+  | protected bot_le (x : WithBot α) : WithBot.LE ⊥ x
+  | protected coe_le_coe {a b : α} : a ≤ b → WithBot.LE a b
```

Comment:
An alternative approach I tried (but failed to fix all failures) is to define `le` and `lt` using `match`, so that `WithBot.some x ≤ WithBot.some y` is defeq `x ≤ y`.

Reply:
Personally, my goal is for the two relations not to be foralls. It's really annoying that `intro` sees through `<=` on `ENNReal` for example.

Reply:
Can we postpone this PR until Feb 17? If I don't suggest a working alternative (a PR that builds) until then, then we go forward with merging this one.

Reply:
Yes, no problem. I don't have time to work on this PR in the foreseeable future anyway.

Reply:
This is what I ended up doing